android/resources/all/xml/network_security_config.xml

```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="utf-8"?>
+<network-security-config>
+    <base-config cleartextTrafficPermitted="true">
+        <trust-anchors>
+            <certificates src="system" />
+        </trust-anchors>
+    </base-config>
+    <domain-config cleartextTrafficPermitted="false">
+        <domain includeSubdomains="true">mozilla.org</domain>
+        <domain includeSubdomains="true">firefox.com</domain>
+        <domain includeSubdomains="true">mozgcp.net</domain>
+        <domain includeSubdomains="true">sentry.io</domain>
+        <domain includeSubdomains="true">apple.com</domain>
+        <domain includeSubdomains="true">google.com</domain>
+        <domain includeSubdomains="true">github.io</domain> 
```

Comment:
As discussion on Slack, I feel this is very fragile given it is totally reliant on us _remembering_ to put new domains here as they are used. A few ideas: Could we use wildcards here? Or could we use IP addresses and then provide the loopback address as an exception instead of making the base case `true`? 

I was also going to ask if this is necessary at all, because AFAIK we enforce HTTPS on production for all requests on the C++ side. However turns out I could not find that code to link to, so now I am second guessing it 😅

Reply:
>Or could we use IP addresses and then provide the loopback address as an exception instead of making the base case true? 

Tested that, didnt work. 

>Could we use wildcards here?

No, sorry.

> I feel this is very fragile given it is totally reliant on us remembering to put new domains here as they are used.

It's a layer of security, so the goal of this file is to reduce risk, not to be an impenetrable new sec policy. That would be  `usesCleartextTraffic="false"`

>I was also going to ask if this is necessary at all, because AFAIK we enforce HTTPS on production for all requests on the C++ side. However turns out I could not find that code to link to, so now I am second guessing it 😅

Not all requests originate from c++ tho. Glean on android i.e. uses a kotlin worker to post messages.